src/react/projects/spark-core-react/src/SprkTabs/components/SprkTabsButton/SprkTabsButton.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component, createRef } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+class SprkTabsButton extends Component {
+  constructor(props) {
+    super(props);
+    this.tabBtnRef = createRef();
+  }
+
+  componentDidUpdate() {
+    const { isFocused, tabBtnId } = this.props;
+    if ((isFocused !== undefined) && (isFocused === tabBtnId)) {
+      this.tabBtnRef.current.focus();
+    }
+  }
+
+  render() {
+    const {
+      tabBtnChildren,
+      tabBtnId,
+      tabBtnAddClasses,
+      ariaControls,
+      ariaSelected,
+      isActive,
+      onTabClick,
+      tabBtnDataId,
+      tabBtnAnalytics,
+    } = this.props;
+    return (
+      <button
+        className={classnames(
+          'sprk-c-Tabs__button',
+          tabBtnAddClasses,
+          { 'sprk-c-Tabs__button--active': isActive },
+        )}
+        role="tab"
+        onClick={onTabClick}
+        aria-controls={ariaControls}
+        aria-selected={ariaSelected}
+        id={tabBtnId}
+        type="button"
+        tabIndex={isActive ? undefined : '-1'}
+        ref={this.tabBtnRef}
+        data-id={tabBtnDataId}
+        data-analytics={tabBtnAnalytics}
+      >
+        {tabBtnChildren}
+      </button>
+    );
+  }
+}
+
+SprkTabsButton.defaultProps = {
+  tabBtnChildren: undefined,
+  isActive: false,
+  isFocused: undefined,
+  ariaControls: undefined,
+  tabBtnAddClasses: undefined,
+  ariaSelected: undefined,
+  onTabClick: undefined,
+  tabBtnAnalytics: undefined,
+  tabBtnDataId: undefined,
+};
+
+SprkTabsButton.propTypes = {
+  // The content for the button
+  tabBtnChildren: PropTypes.node,
```

Comment:
should this be required?

Reply:
Yes, it definitely should. Updated, thanks!